sdk/core/azure-core/src/main/java/com/azure/core/http/HttpRequest.java

```diff
@@ -184,7 +210,8 @@ public HttpRequest setBody(String content) {
      */
     public HttpRequest setBody(byte[] content) {
         setContentLength(content.length);
-        return setBody(Mono.fromSupplier(() -> ByteBuffer.wrap(content)).flux());
+        this.body = BinaryData.fromBytes(content);
+        return this;
```

Comment:
Same statement as above where this could just call into the `setBody(BinaryData)` method